lib/src/Pos/Logic/Types.hs

```diff
@@ -92,6 +94,12 @@ data Logic m = Logic
     , recoveryInProgress :: m Bool
 
     , securityParams     :: SecurityParams
+
+      -- Characterize bogus block headers.
+    , checkBlockHeader :: BlockHeader -> m Bool
```

Comment:
As I've already said, I think this is useless. Either delete it or implement something viable, please. `classifyHeaders` is a good candidate for `checkBlockHeader`, though you should extend it to many headers.

Reply:
This is somewhat forward thinking (I don't believe it's used now). There's discussion about having a **very cheap** no-false-negatives validity test for block headers, so that the diffusion layer can drop headers which it's sure are invalid.

Reply:
Forward thinking is good, weeds are bad. It's generally a good idea not to have any functionality that is not used, because it's confusing other developers. They might think it's important and overcomplicate something. It's easy to add these methods later when these checks will be actually implemented.